policies.go

```diff
@@ -984,6 +992,37 @@ func (d *dcAwareRR) Reset()                              {}
 func (d *dcAwareRR) KeyspaceChanged(KeyspaceUpdateEvent) {}
 func (d *dcAwareRR) SetPartitioner(p string)             {}
 
+func (d *dcAwareRR) IsOperational(session *Session) error {
+	if !session.cfg.disableInit && !session.cfg.disableControlConn {
+		var datacenters []string
+		hosts, _, err := session.hostSource.GetHosts()
+		if err != nil {
+			return fmt.Errorf("gocql: unable to check if session is operational: %v", err)
+		}
+		for _, host := range hosts {
+			if !session.cfg.filterHost(host) {
+				datacenters = append(datacenters, host.DataCenter())
+			}
+		}
+
+		found := false
+
+		for _, dc := range datacenters {
+			if dc == d.local {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return fmt.Errorf("gocql: datacenter provided in the policy was not found in the topology")
+		}
+	}
+
+	return nil
+
+}
```

Comment:
@sylwiaszunejko , it is a nit, not necessary to following, but early return is a realy good practice to follow:
```suggestion
	if session.cfg.disableInit || session.cfg.disableControlConn {
		return nil
	}
	var datacenters []string
	hosts, _, err := session.hostSource.GetHosts()
	if err != nil {
		return fmt.Errorf("gocql: unable to check if session is operational: %v", err)
	}
	for _, host := range hosts {
		if !session.cfg.filterHost(host) && host.DataCenter() == d.local {
			// Policy can work properly only if there is at least one host from target DC
			// No need to check host status, since it could be down due to the outage
			// We only need to make sure that policy is not misconfigured with wrong DC
			return nil
		}
	}

	return fmt.Errorf("gocql: dcAwarePolicy: datacenter %s was not found in the topology", d.local)
}
```

Reply:
done